core/graphelier-service/db/mongo.go

```diff
@@ -334,54 +385,69 @@ func (c *Connector) GetTopOfBookByInterval(instrument string, startTimestamp uin
 	if err != nil {
 		return nil, err
 	}
-	log.Tracef("Documents in interval {%d,%d}=%d, keeping 1 in %d", startTimestamp/interval, endTimestamp/interval, count, count/maxCount)
+	log.Tracef("Documents in orderbook interval {%d,%d}=%d, keeping 1 in %d", startTimestamp/interval, endTimestamp/interval, count, count/maxCount)
 
-	// Add $mod comparator to interval_multiple filter
-	filter[1].Value = append(filter[1].Value.(bson.D), bson.E{Key: "$mod", Value: bson.A{count / maxCount, 0}})
+	if maxCount < count {
+		// Add $mod comparator to interval_multiple filter
+		filter[1].Value = append(filter[1].Value.(bson.D), bson.E{Key: "$mod", Value: bson.A{count / maxCount, 0}})
+	
+		results, err = getBigTopBookInterval(&filter, collection, interval)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		orderbook, err := c.GetOrderbook(instrument, startTimestamp)
+		if err != nil {
+			return nil, err
+		}
+		messages, err := c.GetMessagesWithinInterval(instrument, startTimestamp, endTimestamp)
+		if err != nil {
+			return nil, err
+		}
+		x := len(messages) / int(maxCount)
+		if x == 0 {
+			x = 1
+		}
+		log.Tracef("Documents in message interval {%d,%d}=%d, keeping 1 in %d", startTimestamp, endTimestamp, len(messages), x)
```

Comment:
The end of this log does not really make sense, we should not be "keeping 1 in %d" messages as they are all important.